### DISCUSSION TITLE
How can I get full-page screenshots from SeleniumBase test?

### DISCUSSION THREAD

**Discussion by javiergsalcedo** [seleniumbase/SeleniumBase#902] [Q&A] [answered]
I want a full screenshot of the pages where I am doing testing, but they are all cropped  whenever I do screenshots

**Comment by mdmintz** [comment, marked as answer]
Hi @popopojgs , if you run tests using ``--headless`` mode, you'll get the full page screenshot. Otherwise they will be the same size as the browser window.

**Reply by javiergsalcedo** [reply]
There's something that has to be wrong, I tried --headless and with different browser sizes, but the screenshots are almost always the same. Is there another way I could try?

**Reply by mdmintz** [reply]
@popopojgs Try [examples/test_image_saving.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_image_saving.py)
It shows you how to take screenshots of specific sections of web pages.

**Reply by javiergsalcedo** [reply]
Solved!

**Reply by allcaps** [reply]
Take the height of the `body` element and resize the window to match it. Like this:

    selector = "body"
    element = self.wait_for_element_visible(selector)
    height = element.size["height"]
    self.set_window_size(1920, height)
    self.save_element_as_image_file(selector, file_name, folder)

Use headless mode to have unlimited height: `pytest --headless`.

**Reply by dsmabulage** [reply]
> Solved!

Can you explain how you solve this

**Reply by mdmintz** [reply]
See https://github.com/seleniumbase/SeleniumBase/discussions/3362#discussioncomment-11650705
